models/articles.models.js

```diff
@@ -35,81 +35,63 @@ const fetchArticles = (
   }
+  //initialise SQL query
+  let sqlQuery = `
+        SELECT a.author, a.title, a.article_id, a.topic, a.created_at, a.votes, a.article_img_url, count(c.article_id) as comment_count 
+        FROM articles a
+        FULL JOIN comments c
+          ON a.article_id = c.article_id 
+          `;
+
+  const queryValues = [];
+
+  //extend sqlQuery string to accomodate author and topic queries
+  if (author) {
+    sqlQuery += `WHERE a.author = $1 `;
+    queryValues.push(author);
+    if (topic) {
+      sqlQuery += `AND a.topic = $2 `;
+      queryValues.push(topic);
+    }
+  } else if (topic) {
+    sqlQuery += `WHERE a.topic = $1 `;
+    queryValues.push(topic);
+  }
 
-  //find comment counts for each article
-  return db
-    .query(
-      `
-    SELECT article_id, COUNT(*) AS comment_count
-    FROM comments
-
-    GROUP BY article_id
-    `
-    )
-    .then(({ rows }) => {
-      //use this info to create lookup object
-      const comment_counts = rows;
-      const lookupObj = {};
-      comment_counts.forEach((article) => {
-        lookupObj[article.article_id] = Number(article.comment_count);
-      });
-
-      //initialise SQL query
-      let sqlQuery = `
-        SELECT author, title, article_id, topic, created_at, votes, article_img_url 
-        FROM articles `;
-
-      const queryValues = [];
-
-      //extend sqlQuery string to accomodate author and topic queries
-      if (author) {
-        sqlQuery += `WHERE author = $1 `;
-        queryValues.push(author);
-        if (topic) {
-          sqlQuery += `AND topic = $2 `;
-          queryValues.push(topic);
-        }
-      } else if (topic) {
-        sqlQuery += `WHERE topic = $1 `;
-        queryValues.push(topic);
-      }
-
-      //add order by queries to sqlQuery string
-      sqlQuery += `ORDER BY ${sort_by} ${order}`;
-      sqlQuery += `;`;
-
-      //query database again to get the rest of the information required
-      return db.query(sqlQuery, queryValues).then(({ rows }) => {
-        const table = rows;
-        //reject with a 404 if nothing is found
-        if (!table.length) {
-          return Promise.reject({ status: 404, msg: "Not found" });
-        }
+  //add order by and group by queries to sqlQuery string
+  sqlQuery += `GROUP BY a.article_id `;
+  sqlQuery += `ORDER BY ${sort_by} ${order}`;
+  sqlQuery += `;`;
 
-        //use lookup object to append comment counts to each article
-        else {
-          table.forEach((article) => {
-            if (lookupObj[article.article_id]) {
-              article.comment_count = lookupObj[article.article_id];
-            } else {
-              article.comment_count = 0;
-            }
-          });
-          return table;
-        }
-      });
+  //query database to get the information required
+  return db.query(sqlQuery, queryValues).then(({ rows }) => {
+    //convert the comment counts from string to number
+    rows.forEach((row) => {
+      row.comment_count = Number(row.comment_count);
```

Comment:
We can also get comment count as an integer directly from the sql query by putting ::int after you select the value, e.g count(c.article_id)::int - this is called the type casting